### DISCUSSION TITLE
CliLocalTerminalSession with cin  help

### DISCUSSION THREAD

**Discussion by Dovahkiin02** [daniele77/cli#173] [Q&A] [answered]
I'm trying to read an input string from the command line. It works fine with CliFileSession but when I use CliLocalTerminalSession  the text the user types in becomes invisible on the command line. What am I doing wrong?

Here is my code:

```
auto rootMenu = make_unique<Menu>("cli");
std::string input;
rootMenu->Insert("test", [&input](auto out) {
    out << "input: ";
    std::getline(std::cin, input);
}, "get input string from user");

Cli cli(move(rootMenu));

StandaloneAsioScheduler scheduler;
CliLocalTerminalSession localSession(cli, scheduler, std::cout, 200);

scheduler.Run();
```

**Comment by daniele77** [comment, marked as answer]
Hi @Dovahkiin02 ,
the library is not intended to be used with other user input except the one managed by the library itself.
This means that when using CLI library the handlers must not contain blocking instructions such as `std::getline`, `std::getch`, and so on...

**Comment by daniele77** [comment]
I've just committed a fix that should enable user keyboard handling on Linux platform as well (commit 17dafd8).
It's the issue #239.
Now you can use `std::cin` and `std::getline` inside command handlers (local session only, of course).
Please, see if this solves your problem.
